Realm/Swift/RLMSupport.swift

```diff
@@ -101,7 +102,10 @@ extension RLMCollection {
      Returns the index of the first object in the collection matching the predicate.
      */
     public func indexOfObject(where predicateFormat: String, _ args: CVarArg...) -> UInt {
-        return indexOfObject(with: NSPredicate(format: predicateFormat, arguments: getVaList(args)))
+        guard let index = indexOfObject?(with: NSPredicate(format: predicateFormat, arguments: getVaList(args))) else {
```

Comment:
Is this valid syntax?

Reply:
Because `indexOfObject` has been marked optional in `RLMCollection` it requires to be unwrapped. The `?` in the method signature does look funky though.

Reply:
Ah, the method itself is optional. I don't think I've ever seen that before. I guess it's only possible via obj-c protocols so it won't show up much.